src/util/Thread.hpp

```diff
@@ -18,5 +21,399 @@ enum class ThreadPriority {
 };
 
 auto print(ThreadPriority priority) noexcept -> const char*;
+auto SetThisThreadsName(std::string_view threadName) noexcept -> void;
 auto SetThisThreadsPriority(ThreadPriority priority) noexcept -> void;
+
+auto adjustThreadName(
+    std::string_view threadName,
+    std::string&& appender) noexcept -> std::string;
+
+constexpr std::string_view loggerThreadName{"Logger\0"};
+static_assert(
```

Comment:
it's needed here, one line below place where it's hardcoded?

Reply:
it is not needed but it prevents you from making name too long. It will not be an error or whatever but if name will be longer then thread name will not be applied